### PR TITLE
Add "(GPN: gcsfuse-<client-identifier>)" to useragent

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
Added GPN:gcsfuse-<client-identifier> to user agent.
Removed redundant client identifier(appname or metadata image type env variable) from user agent if it has been appended to GPN.

Possible Scenarios and changes:
1. When both GCSFUSE_METADATA_IMAGE_TYPE  environment variable and app-name flag are set
      - `GCSFUSE_METADATA_IMAGE_TYPE=DLC` `app-name=myapp`
      - Old UserAgent = `gcsfuse/0.42.3 (Go version go1.19.5) myapp DLC,gzip(gfe)`
      - New UserAgent =`gcsfuse/0.42.3 (Go version go1.19.5) myapp (GPN:gcsfuse-DLC),gzip(gfe)`
2.  When both GCSFUSE_METADATA_IMAGE_TYPE  environment variable and app-name flag are unset
      - `GCSFUSE_METADATA_IMAGE_TYPE=""` `app-name=""`
      - Old UserAgent = `gcsfuse/0.42.3 (Go version go1.19.5),gzip(gfe)`
      - New UserAgent =`gcsfuse/0.42.3 (Go version go1.19.5) (GPN:gcsfuse),gzip(gfe)`
3. When GCSFUSE_METADATA_IMAGE_TYPE is set and app-name is unset
      - `GCSFUSE_METADATA_IMAGE_TYPE=DLC` `app-name=""`
      - Old UserAgent = `gcsfuse/0.42.3 (Go version go1.19.5) DLC,gzip(gfe)`
      - New UserAgent =`gcsfuse/0.42.3 (Go version go1.19.5) (GPN:gcsfuse-DLC),gzip(gfe)`
4. When GCSFUSE_METADATA_IMAGE_TYPE is unset and app-name is set
      - `GCSFUSE_METADATA_IMAGE_TYPE=""` `app-name=myapp`
      - Old UserAgent = `gcsfuse/0.42.3 (Go version go1.19.5) myapp,gzip(gfe)`
      - New UserAgent =`gcsfuse/0.42.3 (Go version go1.19.5) (GPN:gcsfuse-myapp),gzip(gfe)`

### Testing details
1. Manual - Manually verified that the user agent has GPN appended
7. Unit tests - added
8. Integration tests - NA

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -76,7 +76,14 @@ func registerSIGINTHandler(mountPoint string) {
 }
 
 func getUserAgent(appName string) string {
-	return strings.TrimSpace(fmt.Sprintf("gcsfuse/%s %s %s", getVersion(), appName, os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE")))
+	if len(os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE")) > 0 {
```
Comment (sethiay) [inline main.go:79]: We are calling os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE") twice. So can we create variable and use it at both places ?

File: main.go [hunk 2]
```diff
@@ -76,7 +76,14 @@ func registerSIGINTHandler(mountPoint string) {
...
+		userAgent := fmt.Sprintf("gcsfuse/%s %s (GPN:gcsfuse-%s)", getVersion(), appName, os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE"))
+		return strings.Join(strings.Fields(userAgent), " ")
```
Comment (sethiay) [inline main.go:81]: why do we need strings.Join here ?

File: internal/storage/storage_handle_test.go [hunk 3]
```diff
@@ -116,10 +116,3 @@ func (t *StorageHandleTest) TestNewStorageHandleWithZeroMaxConnsPerHost() {
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
-
-func (t *StorageHandleTest) TestNewStorageHandleWhenUserAgentIsSet() {
-	sc := getDefaultStorageClientConfig()
-	sc.UserAgent = "gcsfuse/unknown (Go version go1.20-pre3 cl/474093167 +a813be86df)"
```
Comment (sethiay) [inline internal/storage/storage_handle_test.go:122]: you can set different value of sc.UserAgent (other than the default) and keep this test case?

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -44,6 +44,6 @@ jobs:
     - name: golangci-lint
       uses: golangci/golangci-lint-action@032fa5c5e48499f06cf9d32c02149bfac1284239
       with:
-        args: -E=goimports
+        args: -E=goimports --timeout 2m0s
```
Comment (sethiay) [inline .github/workflows/ci.yml:47]: Thanks, we should put this change in a separate PR.

**Review by sethiay — COMMENTED** [review]
Please add examples of final user agent string in each of the 4 combinations possible with app-name and env variable in the Description of this PR.

File: main.go (hunk 2 above)
Comment (ashmeenkaur) [reply main.go:81]: Line 81 is to remove any redundant spaces from the string incase appname is not present. strings.Fields will split the string around spaces and strings.Join will join the fields into a string.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (ashmeenkaur) [reply .github/workflows/ci.yml:47]: Raised a separate PR: https://github.com/GoogleCloudPlatform/gcsfuse/pull/1113

File: internal/storage/storage_handle_test.go (hunk 3 above)
Comment (ashmeenkaur) [reply internal/storage/storage_handle_test.go:122]: Added, thanks!

File: main.go (hunk 1 above)
Comment (ashmeenkaur) [reply main.go]: Done, thanks!

**Review by vadlakondaswetha — APPROVED** [review]

**Review by sethiay — APPROVED** [review]
